### DISCUSSION TITLE
Are cordova versions old?

### DISCUSSION THREAD

**Discussion by realtebo** [quasarframework/quasar#15181] [CLI - Capacitor / Cordova modes] [answered]
I created a fresh new project

I added android platform
Then I added cordova-plugin-file using cordova cli

I see this in `package.json` 

```
  "devDependencies": {
    "cordova-android": "^10.1.2",
    "cordova-plugin-file": "^7.0.0"
  },
```

Why are being added an old version of cordova-android? (actualy we are at v11)
And also, why a SO OLD of cordova-plugin-file!?! Actual version is 10.

Am i doing something wrong?

**Comment by rstoenescu** [comment, marked as answer]
Upgrade your cordova CLI version then re-add the Quasar Cordova mode.
Quasar CLI relies on your globally installed cordova cli.

**Reply by realtebo** [reply]
Not enough.
This doesn't fix the "problem" of cordova-android being not the last
And anyway last cordova-cli installs an old version of cordova-plugin-file and I do not know if it depends from old version of cordova-android or what else.

**Comment by realtebo** [comment]
I installed cordova cli just a few days ago. could not be old.
```

 cordova --v
11.0.0
```

I tried to remove cordova mode

```
quasar mode remove android

 App • ⚠️  Unknown mode "android" to remove

C:\laragon\www\offline> quasar mode remove cordova

? Will also remove /src-cordova folder. Are you sure? Yes
 App • Cordova support was **removed**
```

I upgraded quasar itself

```
quasar upgrade -i         

  Quasar CLI · Gathering information with yarn...

 @quasar/extras: 1.15.8 → 1.15.9
 quasar: 2.11.2 → 2.11.3
```


Then I re-add `cordova` mode

```
quasar mode add cordova
quasar dev -m cordova -T android
 App • [sync] Running "cordova platform add android" in C:\laragon\www\offline\src-cordova

Using cordova-fetch for cordova-android@^10.1.1
...
Android project created with cordova-android@10.1.2
```

The I used the cordova cli to add the plugin

```
cd .\src-cordova\
C:\laragon\www\offline\src-cordova> cordova plugin add  cordova-plugin-file
```

The result is 
```
"devDependencies": {
    "cordova-android": "^10.1.2",
    "cordova-plugin-file": "^7.0.0"
  },
```

**Reply by metalsadman** [reply]
upgrade it yourself https://cordova.apache.org/announcements/2022/07/12/cordova-android-release-11.0.0.html
it's the cordova cli that's not installing the latest cordova-android version, you can see it running `cordova platform`.
![image](https://user-images.githubusercontent.com/31038838/210177247-46029807-8fd3-49fa-a49f-c283065b85cb.png)

**Reply by metalsadman** [reply]
cordova plugin file isn't at version 10, see here is just version 7 https://www.npmjs.com/package/cordova-plugin-file
it's your job to install what plugins you are using but don't update them blindly without reading release notes/docs/requirements etc..
